controllers/courses.php

```diff
@@ -14,31 +14,31 @@ function index()
 
         $sql = "SELECT *
                 FROM period_courses
-                  natural join `year`
-                  natural join course
-                  natural join subject
-                  natural join period
-                  natural join `group`
-                  natural join task
-                WHERE course.person_id = $my_id";
+                  join courses USING (course_id)
+                  join subjects USING (subject_id)
+                  join periods USING (period_id)
+                  join years USING (year_id)
+                  join groups USING (group_id)
+                  natural join tasks 
+                WHERE courses.teacher_id = $my_id";
         $this->period_courses = get_all($sql);
-        $this->periods = get_all("SELECT * FROM period");
-        $this->taught_courses = get_all("SELECT * FROM course NATURAL JOIN subject NATURAL JOIN `group` WHERE `course`.person_id={$my_id}");
+        $this->periods = get_all("SELECT * FROM periods");
+        $this->taught_courses = get_all("SELECT * FROM courses NATURAL JOIN subjects NATURAL JOIN groups WHERE courses.teacher_id={$my_id}");
         $this->courses_taken = get_all("SELECT * FROM group_persons gp
-                                                 JOIN course c ON c.group_id = gp.group_id
-                                                 JOIN `group` g ON g.group_id = c.group_id
-                                                 JOIN `subject` s ON s.subject_id = c.subject_id
-                                                 JOIN `person` p ON p.person_id = c.person_id
+                                                 JOIN courses c ON c.group_id = gp.group_id
+                                                 JOIN groups g ON g.group_id = c.group_id
+                                                 JOIN subjects s ON s.subject_id = c.subject_id
+                                                 JOIN persons p ON p.person_id = c.teacher_id
                                                  WHERE `gp`.person_id={$my_id}");
 
     }
 
     private function get_current_period_id()
     {
-        return get_one("SELECT period_id FROM period WHERE now() BETWEEN period_start AND period_end");
+        return get_one("SELECT period_id FROM periods WHERE now() BETWEEN period_start AND period_end");
     }
     function view() {
-        $this->course = get_first("SELECT * FROM course NATURAL JOIN person NATURAL JOIN subject NATURAL JOIN `group` WHERE course_id = '{$this->params[0]}'");
-        $this->task = get_first("SELECT * FROM task");
+        $this->course = get_first("SELECT * FROM courses NATURAL JOIN persons NATURAL JOIN subjects NATURAL JOIN groups WHERE course_id = '{$this->params[0]}'");
```

Comment:
Peaks tehtud olema

2016-03-21 13:15 GMT+02:00 henno <notifications@github.com>:

> In controllers/courses.php
> <https://github.com/henno/teodor/pull/18#discussion_r56807241>:
>
> >      }
> >      function view() {
> > -        $this->course = get_first("SELECT * FROM course NATURAL JOIN person NATURAL JOIN subject NATURAL JOIN `group` WHERE course_id = '{$this->params[0]}'");
> > -        $this->task = get_first("SELECT * FROM task");
> > +        $this->course = get_first("SELECT * FROM courses NATURAL JOIN persons NATURAL JOIN subjects NATURAL JOIN groups WHERE course_id = '{$this->params[0]}'");
>
> Siin muutmata (NATURAL)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/henno/teodor/pull/18/files/0b7c7f24b36599cfba2d1838f593a733ea401c09#r56807241>
>



-- 
Lugupidamisega,
Rasmus Reimaa
VS15